Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -179,409 +135,356 @@ public void Dispose()
+                    }
+                }
 
-                if (MatchesPublishedServerUrl(source, isExternal, out string res, out port))
+                // Remove all IPv4 interfaces if IPv4 is disabled
+                if (!IsIpv4Enabled)
                 {
-                    _logger.LogDebug("{Source}: Using BindAddress {Address}:{Port}", source, res, port);
-                    return res;
+                    _interfaces.RemoveAll(x => x.AddressFamily == AddressFamily.InterNetwork);
                 }
-            }
-
-            _logger.LogDebug("GetBindInterface: Source: {HaveSource}, External: {IsExternal}:", haveSource, isExternal);
 
-            // No preference given, so move on to bind addresses.
-            if (MatchesBindInterface(source, isExternal, out string result))
-            {
-                return result;
-            }
+                // Remove all IPv6 interfaces if IPv6 is disabled
+                if (!IsIpv6Enabled)
+                {
+                    _interfaces.RemoveAll(x => x.AddressFamily == AddressFamily.InterNetworkV6);
+                }
 
-            if (isExternal && MatchesExternalInterface(source, out result))
-            {
-                return result;
+                _logger.LogInformation("Using bind addresses: {0}", _interfaces.OrderByDescending(x => x.AddressFamily == AddressFamily.InterNetwork).Select(x => x.Address));
             }
+        }
 
-            // Get the first LAN interface address that isn't a loopback.
-            var interfaces = CreateCollection(
-                _interfaceAddresses
-                    .Exclude(_bindExclusions, false)
-                    .Where(IsInLocalNetwork)
-                    .OrderBy(p => p.Tag));
-
-            if (interfaces.Count > 0)
+        /// <summary>
+        /// Initialises the remote address values.
+        /// </summary>
+        private void InitialiseRemote(NetworkConfiguration config)
+        {
+            lock (_initLock)
             {
-                if (haveSource)
+                // Parse config values into filter collection
+                var remoteIPFilter = config.RemoteIPFilter;
+                if (remoteIPFilter.Any() && !string.IsNullOrWhiteSpace(remoteIPFilter.First()))
                 {
-                    foreach (var intf in interfaces)
-                    {
-                        if (intf.Address.Equals(source.Address))
-                        {
-                            result = FormatIP6String(intf.Address);
-                            _logger.LogDebug("{Source}: GetBindInterface: Has found matching interface. {Result}", source, result);
-                            return result;
-                        }
-                    }
+                    // Parse all IPs with netmask to a subnet
+                    _ = NetworkExtensions.TryParseToSubnets(remoteIPFilter.Where(x => x.Contains('/', StringComparison.OrdinalIgnoreCase)).ToArray(), out _remoteAddressFilter, false);
```

Comment:
Why discard?

Reply:
You should also put `remoteIPFilter.Where(x => x.Contains('/', StringComparison.OrdinalIgnoreCase)).ToArray()` in a variable